frontend/src/modules/vehicles/AddToVehicleList.vue

```diff
@@ -0,0 +1,98 @@
+<template>
+  <div>
+    <CCol>
+      <CAlert v-if="!!errorMessage" color="danger">{{ errorMessage }}</CAlert>
+      <CSelect 
+        v-if="!!userVehicleLists"
+        :value.sync="selectedIndex"
+        placeholder="Select a vehicle list"
+        :options="options"
+      />
+      <CRow class="d-flex justify-content-center">
+        <CButton color="primary" class="mr-2" :disabled="addBtnDisabled" @click="validateList">Add</CButton>
+        <CButton color="danger" @click="closeModal">Cancel</CButton>
+      </CRow>
+    </CCol>
+  </div>
+</template>
+
+<script>
+const axios = require('axios');
+
+export default {
+  props: ['vehicleId', 'closeModal', 'showMessage'],
+  data() {
+    return {
+      userVehicleLists: null,
+      selectedIndex: -1,
+      options: null,
+      addBtnDisabled: false,
+      errorMessage: null
+    }
+  },
+  methods: {
+    validateList() {
+      console.log(this.selectedIndex);
+      if (this.selectedIndex < 0) {
+        this.showErrorMessage('You must select a vehicle list to add the vehicle to.');
+      } else if (this.userVehicleLists[this.selectedIndex].vehicles.includes(this.vehicleId)) {
+        this.showErrorMessage('This vehicle already exists in this list.');
+      } else {
+        this.addVehicleToList(this.userVehicleLists[this.selectedIndex]._id);
+      }
+    },
+    showErrorMessage(message) {
+      this.errorMessage = message;
+      setTimeout(() => {
+        this.errorMessage = null;
+      }, 5000);
+    },
+    addVehicleToList(vehicleListId) {
+      console.log("test");
+      axios({
+        method: "POST",
+        url: `${this.$store.state.api}/vehicle-list/${vehicleListId}/add`,
+        headers: {
+          Authorization: `Bearer ${this.$store.state.auth.token}`,
+        },
+        data: {
+          vehicles: [this.vehicleId]
+        }
+      }).then((response) => {
+        this.showMessage(`The vehicle has been added to the list "${response.data.payload.title}"`, "success");
+        this.closeModal();
+      }).catch((error) => {
+        console.log(error);
+      });
+    },
+    fetchUserVehicleLists() {
+      axios({
+        method: "GET",
+        url: `${this.$store.state.api}/vehicle-list/user/${this.$store.state.auth.userId}`,
+        headers: {
+          Authorization: `Bearer ${this.$store.state.auth.token}`,
+        },
+      }).then((response) => {
+        this.userVehicleLists = response.data.payload;
+        let options = [];
+        this.userVehicleLists.forEach((list, index) => {
+          options.push({
+            value: index,
+            label: list.title
+          });
+        });
+        this.options = options;
+      }).catch((error) => {
+        console.log(error);
+      });
+    }
+  },
+  beforeMount() {
+    this.fetchUserVehicleLists();
+  }
+}
+</script>
+
+<style>
+
+</style>
```

Comment:
Empty style block? Really easy fix to remove this, otherwise it's fine.

Reply:
Resolved in this [commit](https://github.com/purplesmurf1998/Autotracks/pull/62/commits/9ae2e8c3142594aafefadebf4921df6fe6e157d0)